app/src/main/java/com/dragic/gamehunter/view/home/HomeDeals.kt

```diff
@@ -61,4 +67,20 @@ fun HomeDeals(
             )
         }
     }
+
+    if (deals.isNotEmpty()) {
```

Comment:
It would be better to check `deals.isNotEmpty()` in the ViewModel. As we discussed, the View should only notify the ViewModel of certain events. The ViewModel should parse the events and determine what to do with them. If another page is currently loading and a load more request is received, it should know to ignore it. 

Similarly, if the user scrolls to the end of an empty page and requests more data, this could be interpreted in multiple ways. We could ignore it if the first page is currently loading, or retry the request for the first page if it failed previously. Try implementing these cases for exercise.